sound/soc/sof/pcm.c

```diff
@@ -566,31 +566,6 @@ static int sof_pcm_new(struct snd_soc_pcm_runtime *rtd)
 	return ret;
 }
 
-static void sof_pcm_free(struct snd_pcm *pcm)
-{
-	struct snd_sof_pcm *spcm;
-	struct snd_soc_pcm_runtime *rtd = pcm->private_data;
-	struct snd_soc_component *component =
-		snd_soc_rtdcom_lookup(rtd, DRV_NAME);
-	struct snd_sof_dev *sdev =
-		snd_soc_component_get_drvdata(component);
-
-	spcm = snd_sof_find_spcm_dai(sdev, rtd);
-	if (!spcm) {
-		dev_warn(sdev->dev, "warn: can't find PCM with DAI ID %d\n",
-			 rtd->dai_link->id);
-		return;
-	}
-
-	if (spcm->pcm.playback)
-		snd_dma_free_pages(&spcm->stream[SNDRV_PCM_STREAM_PLAYBACK].page_table);
-
-	if (spcm->pcm.capture)
-		snd_dma_free_pages(&spcm->stream[SNDRV_PCM_STREAM_CAPTURE].page_table);
-
-	snd_sof_free_topology(sdev);
-}
-
```

Comment:
Should we keep ```sof_pcm_free```? Please see https://github.com/thesofproject/linux/pull/335/commits/cbdc405a0ec32dd3ae7efb2f9b3e2739026e2984. We still need to call ```snd_pcm_lib_preallocate_free_for_all``` in  ```sof_pcm_free```

Reply:
or let's allocate everything in dai_load, but not do a half-baked job in two places.

Reply:
> Should we keep `sof_pcm_free`? Please see [cbdc405](https://github.com/thesofproject/linux/commit/cbdc405a0ec32dd3ae7efb2f9b3e2739026e2984). We still need to call `snd_pcm_lib_preallocate_free_for_all` in `sof_pcm_free`

@bardliao doesnt snd_pcm_free() already call this? Do we need to call it again in sof_pcm_free?

Reply:
@ranj063 You are right. We don't need to call it in our driver. @plbossart I believe that we don't need snd_pcm_lib_preallocate_free_for_all() see https://patchwork.kernel.org/patch/6854731/
I vote for allocating everything in dai_load.